### ISSUE TITLE
[Discussion] Ideas for Launcher UI

### ISSUE THREAD

**Issue by Starystars67** [BeamMP/BeamMP-Launcher#65]
I created this discussion to see what the interest is and would be for the Launcher having a GUI rather than a CLI.
Please feel free to drop your ideas and suggestions below too.

- Microphone Selection & Accompanying Enable / Disable Setting
- Nice to use UI
- Login & Account Management
- Interaction with resources folder (E.g. Show the size, button to clear, set a restriction on the maximum size)
- Change log
- Theming for seasons (nice to have)
- Events & News
- Private Server Creation (Using TCP holepunch to remove need for port forwarding? Similar to steam servers)
- Friend inviting system

**Comment by O1LER** [comment]
- Login & Account Management, would be nice to have, logging in before starting beamNG
- Interaction with resources folder (E.g. Show the size, button to clear, set a restriction on the maximum size), also change location
 I think those are the most essential
 
 Nice to have would be 
 - Nice to use UI
 - Change log
- Theming for seasons (nice to have)
- Events & News

Really cool would be proximity voice chat, and with that the microphone stuff mentioned.

I guess steam integration is too complicated? Like logging in with your steam account, joining a friends game and all that?

**Comment by lionkor** [comment]
I think the UI idea is good, but I think it should get low prio. However, nothing wrong with using this time to feel out what people would like in a UI.

It doesnt have to be crossplatform, because the launcher isnt.
It should, however, be small and fast. Native UIs would come to my mind, and I dont see the need for theming as the launcher should get out of everyone's way.

**Comment by SaltySnail** [comment]
Restart server for mod debugging, saves a few inputs.
